esphome/components/toshiba/toshiba.cpp

```diff
@@ -97,6 +99,300 @@ const std::vector<uint8_t> RAC_PT1411HWRU_TEMPERATURE_F{0x10, 0x30, 0x00, 0x20,
                                                         0x22, 0x06, 0x26, 0x07, 0x05, 0x25, 0x04, 0x24, 0x0C,
                                                         0x2C, 0x0D, 0x2D, 0x09, 0x08, 0x28, 0x0A, 0x2A, 0x0B};
 
+// RAS-2819T protocol constants
+const uint16_t RAS_2819T_HEADER1 = 0xC23D;
+const uint8_t RAS_2819T_HEADER2 = 0xD5;
+const uint8_t RAS_2819T_MESSAGE_LENGTH = 6;
+
+// RAS-2819T fan speed codes for rc_code_1 (bytes 2-3)
+const uint16_t RAS_2819T_FAN_AUTO = 0xBF40;
+const uint16_t RAS_2819T_FAN_QUIET = 0xFF00;
+const uint16_t RAS_2819T_FAN_LOW = 0x9F60;
+const uint16_t RAS_2819T_FAN_MEDIUM = 0x5FA0;
+const uint16_t RAS_2819T_FAN_HIGH = 0x3FC0;
+
+// RAS-2819T fan speed codes for rc_code_2 (byte 1)
+const uint8_t RAS_2819T_FAN2_AUTO = 0x66;
+const uint8_t RAS_2819T_FAN2_QUIET = 0x01;
+const uint8_t RAS_2819T_FAN2_LOW = 0x28;
+const uint8_t RAS_2819T_FAN2_MEDIUM = 0x3C;
+const uint8_t RAS_2819T_FAN2_HIGH = 0x50;
+
+// RAS-2819T second packet suffix bytes for rc_code_2 (bytes 3-5)
+// These are fixed patterns, not actual checksums
+struct Ras2819tPacketSuffix {
+  uint8_t byte3;
+  uint8_t byte4;
+  uint8_t byte5;
+};
+const Ras2819tPacketSuffix RAS_2819T_SUFFIX_AUTO{0x00, 0x02, 0x3D};
+const Ras2819tPacketSuffix RAS_2819T_SUFFIX_QUIET{0x00, 0x02, 0xD8};
+const Ras2819tPacketSuffix RAS_2819T_SUFFIX_LOW{0x00, 0x02, 0xFF};
+const Ras2819tPacketSuffix RAS_2819T_SUFFIX_MEDIUM{0x00, 0x02, 0x13};
+const Ras2819tPacketSuffix RAS_2819T_SUFFIX_HIGH{0x00, 0x02, 0x27};
+
+// RAS-2819T swing toggle command
+const uint64_t RAS_2819T_SWING_TOGGLE = 0xC23D6B94E01F;
+
+// RAS-2819T single-packet commands
+const uint64_t RAS_2819T_POWER_OFF_COMMAND = 0xC23D7B84E01F;
+
+// RAS-2819T known valid command patterns for validation
+const std::vector<uint64_t> RAS_2819T_VALID_SINGLE_COMMANDS = {
```

Comment:
Use `std::array` for fixed-size arrays. It consumes less memory (both RAM and flash).
```suggestion
const std::array<uint64_t, 2> RAS_2819T_VALID_SINGLE_COMMANDS = {
```